src/Vision/VisionHelpersTrait.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Vision;
+
+use Google\ApiCore\ArrayTrait;
+use InvalidArgumentException;
+
+/**
+ * Provides helper methods for generated Vision clients.
+ */
+trait VisionHelpersTrait
+{
+    use ArrayTrait;
+
+    /**
+     * A list of allowed url schemes.
+     *
+     * @var array
+     */
+    private $urlSchemes = [
+        'http',
+        'https',
+        'gs'
+    ];
+
+    private function annotateImageHelper(
+        $callback,
+        $requestClass,
+        $image,
+        $features,
+        $optionalArgs = []
+    ) {
+
+        $request = new $requestClass();
+        $request->setImage($image);
+        $request->setFeatures($features);
+        $imageContext = $this->pluck('imageContext', $optionalArgs, false);
+        if (!is_null($imageContext)) {
+            $request->setImageContext($imageContext);
+        }
+        return $callback([$request], $optionalArgs)->getResponses()[0];
+    }
+
+    private function buildFeatureList(
```

Comment:
This signature would fix on a single line.

Reply:
also, docblocks on trait methods, even if private, are useful! :)

Reply:
Done and done